src/backends/github/API.js

```diff
@@ -15,6 +15,7 @@ export default class API {
     this.branch = config.branch || "master";
     this.repo = config.repo || "";
     this.repoURL = `/repos/${ this.repo }`;
+    this.branchNameMaxLength = (config.backend && config.backend.branchNameMaxLength) || 40;
```

Comment:
Did you actually test this line? I'm not sure how it works, seeing that `config` is not the actual config.yml, only the values that are passed in here: https://github.com/netlify/netlify-cms/blob/ca1ecc5fc04afdaad4726c2d0eb71e402f6e19a9/src/backends/github/implementation.js#L33

Reply:
Yeah we need to start rooting out ambiguous variable names like this, there's a lot of it in our source.